src/LeanCode.ContractsGenerator/Compilation/ContractTypes.cs

```diff
@@ -222,4 +230,25 @@ public bool IsReadOnlyDictionary(ITypeSymbol i)
                 || Dictionary.Contains(ns.ConstructUnboundGenericType())
             );
     }
+
+    public bool IsRecordEquatable(ITypeSymbol i)
+    {
+        return i is INamedTypeSymbol ns
+            && ns.IsGenericType
+            && ns.TypeArguments.FirstOrDefault() is INamedTypeSymbol namedType
+            && namedType.IsRecord
+            && Equatable.Contains(ns.ConstructUnboundGenericType());
+    }
+
+    /// <remarks>
+    /// `ContainingType` is not accessible from an `ITypeSymbol`.
+    /// Hence, passing the `IPropertySymbol`.
+    /// </remarks>
+    public bool IsRecordEqualityContract(IPropertySymbol i)
+    {
+        return i.ContainingType is INamedTypeSymbol ns
+            && ns.IsRecord
+            && i.Name == RecordEqualityContractPropertyName
+            && Type.Contains(i.Type);
```

Comment:
What does it give us? If the parent is a record, then `EqualityContract` property name is reserved (or we can treat it like that), no matter what type it is.